From 46c4e89c27b9a6b434d74ce1e53aa5ad312edf16 Mon Sep 17 00:00:00 2001
From: "sameer.mulla" <sameer.mulla@lge.com>
Date: Thu, 15 Nov 2018 11:08:44 +0530
Subject: [PATCH] Use system bus instead of session for obexd

:Release Notes:
Use system bus instead of session for obexd

:Detailed Notes:
Multi user is not supported in webos rpi
so using system bus instead of session bus for obexd

:Testing Performed:
Builded and Tested
Tested with obexctl command line tool, all commands are working fine

:QA Notes:

:Issues Addressed:
[PLAT-66305] Modify bluez to use system bus instead of session bus
             for obexd
---
Upstream-Status: Inappropriate [webos specific]

 obexd/client/ftp.c      |  3 ++-
 obexd/client/map.c      |  2 +-
 obexd/client/opp.c      |  2 +-
 obexd/client/pbap.c     |  2 +-
 obexd/client/session.c  |  2 +-
 obexd/client/sync.c     |  2 +-
 obexd/plugins/pcsuite.c |  2 +-
 obexd/src/manager.c     |  2 +-
 src/bluetooth.conf      | 12 ++++++++++++
 tools/obexctl.c         |  2 +-
 10 files changed, 22 insertions(+), 9 deletions(-)

diff --git a/obexd/client/ftp.c b/obexd/client/ftp.c
index 160e0636a..f492965b2 100644
--- a/obexd/client/ftp.c
+++ b/obexd/client/ftp.c
@@ -463,7 +463,8 @@ int ftp_init(void)
 
 	DBG("");
 
-	conn = dbus_bus_get(DBUS_BUS_SESSION, NULL);
+	conn = dbus_bus_get(DBUS_BUS_SYSTEM, NULL);
+
 	if (!conn)
 		return -EIO;
 
diff --git a/obexd/client/map.c b/obexd/client/map.c
index 74828cddb..ea3d25629 100644
--- a/obexd/client/map.c
+++ b/obexd/client/map.c
@@ -2063,7 +2063,7 @@ int map_init(void)
 
 	DBG("");
 
-	conn = dbus_bus_get(DBUS_BUS_SESSION, NULL);
+	conn = dbus_bus_get(DBUS_BUS_SYSTEM, NULL);
 	if (!conn)
 		return -EIO;
 
diff --git a/obexd/client/opp.c b/obexd/client/opp.c
index 90d0c0c8e..369e9e91c 100644
--- a/obexd/client/opp.c
+++ b/obexd/client/opp.c
@@ -178,7 +178,7 @@ int opp_init(void)
 
 	DBG("");
 
-	conn = dbus_bus_get(DBUS_BUS_SESSION, NULL);
+	conn = dbus_bus_get(DBUS_BUS_SYSTEM, NULL);
 	if (!conn)
 		return -EIO;
 
diff --git a/obexd/client/pbap.c b/obexd/client/pbap.c
index 1ed8c68ec..3af55440b 100644
--- a/obexd/client/pbap.c
+++ b/obexd/client/pbap.c
@@ -1302,7 +1302,7 @@ int pbap_init(void)
 
 	DBG("");
 
-	conn = dbus_bus_get(DBUS_BUS_SESSION, NULL);
+	conn = dbus_bus_get(DBUS_BUS_SYSTEM, NULL);
 	if (!conn)
 		return -EIO;
 
diff --git a/obexd/client/session.c b/obexd/client/session.c
index 7d8ebb04e..884013b75 100644
--- a/obexd/client/session.c
+++ b/obexd/client/session.c
@@ -583,7 +583,7 @@ struct obc_session *obc_session_create(const char *source,
 	if (driver == NULL)
 		return NULL;
 
-	conn = dbus_bus_get(DBUS_BUS_SESSION, NULL);
+	conn = dbus_bus_get(DBUS_BUS_SYSTEM, NULL);
 	if (conn == NULL)
 		return NULL;
 
diff --git a/obexd/client/sync.c b/obexd/client/sync.c
index 92faf4434..c1147897c 100644
--- a/obexd/client/sync.c
+++ b/obexd/client/sync.c
@@ -224,7 +224,7 @@ int sync_init(void)
 
 	DBG("");
 
-	conn = dbus_bus_get(DBUS_BUS_SESSION, NULL);
+	conn = dbus_bus_get(DBUS_BUS_SYSTEM, NULL);
 	if (!conn)
 		return -EIO;
 
diff --git a/obexd/plugins/pcsuite.c b/obexd/plugins/pcsuite.c
index f5a9d9ae8..88f412aba 100644
--- a/obexd/plugins/pcsuite.c
+++ b/obexd/plugins/pcsuite.c
@@ -322,7 +322,7 @@ static gboolean send_backup_dbus_message(const char *oper,
 
 	file_size = size ? *size : 0;
 
-	conn = g_dbus_setup_bus(DBUS_BUS_SESSION, NULL, NULL);
+	conn = g_dbus_setup_bus(DBUS_BUS_SYSTEM, NULL, NULL);
 
 	if (conn == NULL)
 		return FALSE;
diff --git a/obexd/src/manager.c b/obexd/src/manager.c
index 01741fe62..f1fa9fa43 100644
--- a/obexd/src/manager.c
+++ b/obexd/src/manager.c
@@ -486,7 +486,7 @@ gboolean manager_init(void)
 
 	dbus_error_init(&err);
 
-	connection = g_dbus_setup_bus(DBUS_BUS_SESSION, OBEXD_SERVICE, &err);
+	connection = g_dbus_setup_bus(DBUS_BUS_SYSTEM, OBEXD_SERVICE, &err);
 	if (connection == NULL) {
 		if (dbus_error_is_set(&err) == TRUE) {
 			fprintf(stderr, "%s\n", err.message);
diff --git a/src/bluetooth.conf b/src/bluetooth.conf
index b6c614908..f8879c8bb 100644
--- a/src/bluetooth.conf
+++ b/src/bluetooth.conf
@@ -21,10 +21,22 @@
     <allow send_interface="org.freedesktop.DBus.ObjectManager"/>
     <allow send_interface="org.freedesktop.DBus.Properties"/>
     <allow send_interface="org.mpris.MediaPlayer2.Player"/>
+    <allow own="org.bluez.obex"/>
+    <allow send_destination="org.bluez.obex"/>
+    <allow send_interface="org.bluez.obex.Agent1"/>
+    <allow send_interface="org.bluez.obex.Client1"/>
+    <allow send_interface="org.bluez.obex.Session1"/>
+    <allow send_interface="org.bluez.obex.Transfer1"/>
+    <allow send_interface="org.bluez.obex.ObjectPush1"/>
+    <allow send_interface="org.bluez.obex.PhonebookAccess1"/>
+    <allow send_interface="org.bluez.obex.Synchronization1"/>
+    <allow send_interface="org.bluez.obex.MessageAccess1"/>
+    <allow send_interface="org.bluez.obex.Message1"/>
   </policy>
 
   <policy context="default">
     <allow send_destination="org.bluez"/>
+    <allow send_destination="org.bluez.obex"/>
   </policy>
 
 </busconfig>
diff --git a/tools/obexctl.c b/tools/obexctl.c
index 56a76915c..8678b86c9 100644
--- a/tools/obexctl.c
+++ b/tools/obexctl.c
@@ -2154,7 +2154,7 @@ int main(int argc, char *argv[])
 	bt_shell_set_menu(&main_menu);
 	bt_shell_set_prompt(PROMPT_OFF);
 
-	dbus_conn = g_dbus_setup_bus(DBUS_BUS_SESSION, NULL, NULL);
+	dbus_conn = g_dbus_setup_bus(DBUS_BUS_SYSTEM, NULL, NULL);
 
 	client = g_dbus_client_new(dbus_conn, "org.bluez.obex",
 							"/org/bluez/obex");
